### PR TITLE
Update README documentation and optional values

### PR DESCRIPTION
**Pull request by JoeChapman**
- update README
- update how router uses settings, and fallbacks
- hide logger instantiation in test mode

sorry this became a bit larger than I had initially intended, but as I began to update the docs, I realised how little of the config was concrete, and therefore it made little sense to simply update the doc.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -16,7 +15,8 @@ module.exports = options => {
 
   const load = (config) => {
     config.routes.forEach((route) => {
-      app.use(router(route, config));
+      config.route = route;
```
Comment (danielepolencic) [inline index.js]: I don't follow. this code is going to overwrite `config.route` for each route. what is it used for?

File: index.js [hunk 2]
```diff
@@ -31,7 +31,7 @@ module.exports = options => {
         let config = getConfig(options);
       }
       return new Promise((resolve, reject) => {
-        if (config.startOnInitialise === false) {
+        if (config.start === false) {
```
Comment (danielepolencic) [inline index.js]: see #4

File: lib/defaults.js [hunk 3]
```diff
@@ -30,7 +30,8 @@ const defaults = {
   },
   session: {
     ttl: process.env.SESSION_TTL || 1800,
-    secret: process.env.SESSION_SECRET || 'changethis'
+    secret: process.env.SESSION_SECRET || 'changethis',
+    key: process.env.SESSION_KEY || 'hod'
```
Comment (danielepolencic) [inline lib/defaults.js]: I'd rename `key` to `name` since the former isn't officially documented.

File: index.js (hunk 1 above)
Comment (JoeChapman) [inline index.js]: there is no `config.route` before this. I added this to simplify the arguments passed to router

File: index.js (hunk 2 above)
Comment (JoeChapman) [inline index.js]: the config is not being generated from the defaults when start is called independently from the initial call to bootstrap

File: index.js (hunk 1 above)
Comment (danielepolencic) [inline index.js]: if `config.route` is temporary and used only by the current route, perhaps we could write `const routeConfig = Object.assign({route}, config)` and pass that as an argument to the route.

File: index.js [hunk 4]
```diff
@@ -3,20 +3,26 @@
...
 const defaults = require('./lib/defaults');
 
-const getConfig = config => Object.assign({}, defaults, config);
+const getConfig = function (options) {
+  let assigned;
+  [].slice.call(arguments).forEach((arg) => {
+    assigned = Object.assign({}, defaults, arg);
+  })
+  return assigned
+}
```
Comment (danielepolencic) [inline index.js]: `assigned` is overwritten for each argument. When you have `getConfig({a: 1}, {b:2 })` then `assigned` ultimately contains only `Object.assign({}, defaults, {b:2})`. The reference to `{a: 1}` is lost.

An alternative is to use something like:

```js
const args = [].slice.call(arguments);
return Object.assign.apply(null, [{}, defaults].concat(args));
```

Comment (JoeChapman) [inline index.js]: good catch. Would you do the honours and make the change please?

File: index.js [hunk 5]
```diff
@@ -3,20 +3,26 @@
...
 
 module.exports = options => {
 
   const load = (config) => {
     config.routes.forEach((route) => {
-      app.use(router(route, config));
+      const routeConfig = Object.assign({route}, config)
+      app.use(router(routeConfig));
```
Comment (danielepolencic) [inline index.js]: In this case `route` is mutated in place with the values coming from `config`. Is this intentional?

File: lib/router.js [hunk 6]
```diff
@@ -6,43 +6,43 @@ const mixins = hof.mixins;
...
+    fields = require(path.resolve(config.caller, config.fields));
+  } catch (err) {
+    fields = err;
   }
 
-  if (commonFields && fields) {
-    fields = Object.assign({}, commonFields, fields);
+  if ((fields instanceof Error) === false) {
+    fields = Object.assign({}, fields, routeFields);
   }
```
Comment (danielepolencic) [inline lib/router.js]: The `if` expression already evaluates to a `boolean`. It can be replaced with `if(!(fields instanceof Error))`

**Comment by danielepolencic** [comment]
Happy to address the comments if you're ok with it.

**Comment by JoeChapman** [comment]
@danielepolencic please feel free to update code/address comments. Trust you can rebase :)

**Comment by JoeChapman** [comment]
@danielepolencic @easternbloc

**Comment by danielepolencic** [comment]
LGTM 👍

**Pull request merged**